controller/src/endpoints/post.py

```diff
@@ -1,37 +1,239 @@
+            
+            location = user.get("location", "").strip()
+
+            # Create new post
+            new_post = {
+                "_user_id": user_id,
+                "description": description,
+                "images_urls": imgur_urls,
+                "timestamp": datetime.utcnow(),
+                "reactions_count": {"likes": 0},
+                "location": location if location else "",
+            }
+            new_post["timestamp"] = new_post["timestamp"].isoformat()
+
+            post_id = queries.create_post(
+                user_id=user_id,
+                description=description,
+                images_urls=imgur_urls,
+                location=new_post["location"],
+                timestamp=new_post["timestamp"],
+            )
+            log.info(f"Post inserted with ID: {post_id}")
+            if not post_id:
+                log.error("Failed to insert post into database.")
+                api.abort(500, "Failed to create post")
+
+            
+            log.info(f"Returning response: {new_post}")
+
+            return {"post": new_post}, 201
+
+        except Exception as e:
+            log.error(f"Unexpected error in POST /posts: {e}")
+            api.abort(500, f"An unexpected error occurred: {str(e)}")
+
+    @jwt_required()
+    @api.doc(
+        description="Modify an existing post by providing its ID.",
+        params={
+            "post_id": {"description": "The ID of the post to modify.", "example": "12345", "required": True},
+            "description": {"description": "Updated text description of the post.", "example": "Updated description."},
+            "images_urls": {"description": "Updated list of image URLs.", "example": ["url1", "url2"]}
+        }
+    )
+    @api.response(200, "Post updated successfully")
+    @api.response(400, "Failed to update post")
+    @api.response(500, "Internal server error")
     def put(self):
-        '''
-        create
-        '''
+        """
+        Modify an existing post
+        """
+        try:
+            log.info("PUT /posts endpoint hit.")
+            data = request.json
+            log.info(f"Received data: {data}")
+            post_id = data['post_id']
+            description = data.get('description')
+            images_urls = data.get('images_urls')
 
-    def patch(self):
-        '''
-        edit
-        '''
+            log.info(f"Updating post {post_id} with description: {description}, images_urls: {images_urls}")
+            if db().modify_post(post_id, description, images_urls):
+                log.info(f"Post {post_id} updated successfully.")
+                return {"message": "Post updated successfully"}, 200
+            log.error(f"Failed to update post {post_id}.")
+            return {"message": "Failed to update post"}, 400
+        except Exception as e:
+            log.error(f"Error in PUT /posts: {e}")
+            return {"message": "Internal server error"}, 500
 
+    @jwt_required()
+    @api.doc(
```

Comment:
Nie jestem pewny czy api.doc z required na True robi walidację. Na pewno api.expect(model, validate=True) robi walidację i jeżeli jest coś nie tak z payloadem to zwraca 400